src/ILCompiler.Compiler/src/Compiler/JitHelper.cs

```diff
@@ -169,29 +169,50 @@ public static void GetEntryPoint(TypeSystemContext context, ReadyToRunHelper id,
                     break;
 
                 case ReadyToRunHelper.Mod:
-                    mangledName = "RhpIMod";
+                    methodDesc = context.GetHelperEntryPoint("MathHelpers", "IMod");
                     break;
                 case ReadyToRunHelper.UMod:
-                    mangledName = "RhpUMod";
+                    if (context.Target.Architecture == TargetArchitecture.ARMEL)
+                        methodDesc = context.GetHelperEntryPoint("MathHelpers", "UMod");
+                    else
+                        mangledName = "RhpUMod";
                     break;
                 case ReadyToRunHelper.ULMod:
-                    mangledName = "RhpULMod";
+                    if (context.Target.Architecture == TargetArchitecture.ARMEL)
```

Comment:
Could you please fix all the remaining helpers here same way as you have fixed IMod? (Sorry if it was not clear - I meant my comment for all these helpers.)

Reply:
@jkotas Sorry for my misunderstanding )